src/vtn/api/event.rs

```diff
@@ -15,71 +13,126 @@ use openadr::wire::target::TargetLabel;
 use openadr::wire::Event;
 
 use crate::api::{AppResponse, ValidatedQuery};
+use crate::data_source::Crud;
 use crate::error::AppError;
 use crate::error::AppError::{NotFound, NotImplemented};
 use crate::state::AppState;
 
+impl Crud<Event> for AppState {
+    type Id = EventId;
+    type NewType = EventContent;
+    type Error = AppError;
+    type Filter = QueryParams;
+
+    async fn create(&self, content: Self::NewType) -> Result<Event, Self::Error> {
+        if let Some(new_event_name) = &content.event_name {
+            if let Some((name, id)) = self
+                .events
+                .read()
+                .await
+                .iter()
+                .filter_map(|(_, p)| {
+                    p.content
+                        .event_name
+                        .clone()
+                        .map(|name| (name, p.id.clone()))
+                })
+                .find(|(name, _)| name == new_event_name)
+            {
+                warn!(id=%id, event_name=%name, "Conflicting event_name");
+                return Err(AppError::Conflict(format!(
```

Comment:
Event names don't need to be unique, in contrast to program names which have to be unique.

Reply:
ok, pretty sure I did not write that code, just moved it, but we can fix it now